src/mavedb/routers/score_sets.py

```diff
@@ -103,6 +104,17 @@ async def fetch_score_set_by_urn(
         raise HTTPException(status_code=404, detail=f"score set with URN '{urn}' not found")
 
     assert_permission(user, item, Action.READ)
+    if(
+        item
+        and item.superseding_score_set
+        and not owner_or_contributor
+        and (
+            urn_re.MAVEDB_OLD_TMP_URN_RE.fullmatch(item.superseding_score_set.urn)
+            or urn_re.MAVEDB_TMP_URN_RE.fullmatch(item.superseding_score_set.urn)
+        )
+    ):
+        item.superseding_score_set = None
```

Comment:
What if instead of checking like this we check the permissions of the superseding score set, something like:
```suggestion
    from mavedb.lib.permissions import Action, assert_permission, has_permission
    
    if item.superseding_score_set:
        superseding_score_set =  db.scalars(select(ScoreSet).where(ScoreSet.urn = item.superseding_score_set).one()
        
        if not has_permission(user, item, Action.READ):
            item.superseding_score_set = None
```

This way, we don't have to maintain any extra permission logic and can guarantee the item within the superseding score set property is only returned if the user has access to it.

Reply:
I modified it. I check `item.superseding_score_set` directly cause it's an object.